cmd/allocator/main.go

```diff
@@ -193,7 +194,7 @@ func newServiceHandler(kubeClient kubernetes.Interface, agonesClient versioned.I
 		logger.WithError(err).Fatal("starting allocator failed.")
 	}
 
-	if !h.mTLSDisabled {
+	if !h.tlsDisabled {
```

Comment:
getCACertPool(..) is for client auth. 
readTLSCert(..) is for server auth.
So if mTLSDisabled is set, only `caCertPool` refresh is disabled and not `tlsCert`.

Reply:
Done! I also changed the watcher logic to only conditionally watch the cert/tls dirs.